sim/hunter/item_sets_pve.go

```diff
@@ -191,3 +191,170 @@ var ItemSetGiantstalkerPursuit = core.NewItemSet(core.ItemSet{
+						maxSpell := hunter.RaptorStrike
+
+						for _, strike := range hunter.Strikes {
+							if strike.TimeToReady(sim) > maxSpell.TimeToReady(sim) {
+								maxSpell = strike
+							}
+						}
+
+						maxSpell.CD.Reset()
+						aura.Activate(sim) // used for metrics
+					}
+				},
+			}))
+		},
+	},
+})
+
+var ItemSetDragonstalkerPursuit = core.NewItemSet(core.ItemSet{
+	Name: "Dragonstalker's Pursuit",
+	Bonuses: map[int32]core.ApplyEffect{
+		// Your Aimed Shot deals 20% more damage to targets afflicted by one of your trap effects.
+		2: func(agent core.Agent) {
+			hunter := agent.(HunterAgent).GetHunter()
+
+			procAura := hunter.RegisterAura(core.Aura{
+				Label: "S03 - Item - T2 - Hunter - Ranged 2P Bonus",
+				ActionID: core.ActionID{SpellID: 467235},
+				Duration: time.Second * 1, // Duration doesn't really matter as the duration is set to the duration of the trap debuff that applies this aura
+				OnGain: func(aura *core.Aura, sim *core.Simulation) {
+					hunter.AimedShot.DamageMultiplier *= 1.20
+				},
+				OnExpire: func(aura *core.Aura, sim *core.Simulation) {
+					hunter.AimedShot.DamageMultiplier /= 1.20
+				},
+			})
+			
+			core.MakePermanent(hunter.RegisterAura(core.Aura{
+				Label: "S03 - Item - T2 - Hunter - Ranged 2P Bonus Trigger",
+				OnSpellHitDealt: func(aura *core.Aura, sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
+					if spell.Flags.Matches(SpellFlagTrap) {
+						procAura.Duration = spell.DotOrAOEDot(result.Target).Duration
+						procAura.Activate(sim)
+					}
+				},
+			}))
+		},
+		// Your damaging Shot abilities deal 10% increased damage if the previous damaging Shot used was different than the current one.
+		4: func(agent core.Agent) {
+			hunter := agent.(HunterAgent).GetHunter()
+
+			procAura := hunter.RegisterAura(core.Aura{
+				ActionID: core.ActionID{SpellID: 467312},
+				Label:    "S03 - Item - T2 - Hunter - Ranged 4P Bonus",
+				Duration: time.Second * 12,
+
+				OnGain: func(aura *core.Aura, sim *core.Simulation) {
+					for _, spell := range hunter.Shots {
+						if spell != nil && spell != hunter.LastShot {
+							spell.DamageMultiplier *= 1.10
+						}
+					}
+				},
+				OnExpire: func(aura *core.Aura, sim *core.Simulation) {
+					for _, spell := range hunter.Shots {
+						if spell != nil && spell != hunter.LastShot {
+							spell.DamageMultiplier /= 1.10
+						}
+					}
+				},
+			})
+
+			core.MakePermanent(hunter.RegisterAura(core.Aura{
+				Label: "S03 - Item - T2 - Hunter - Ranged 4P Bonus Trigger",
+				OnCastComplete: func(_ *core.Aura, sim *core.Simulation, spell *core.Spell) {
+					if spell.Flags.Matches(SpellFlagShot) {
+						procAura.Deactivate(sim)
+						hunter.LastShot = spell
+						procAura.Activate(sim)
+					}
+				},
+			}))
+		},
+		//  Your Serpent Sting damage is increased by 25% of your Attack Power over its normal duration.
+		6: func(agent core.Agent) {
+			// Implemented in serpent_sting.go
```

Comment:
How you could set this up here is by creating a `serpentStingPctAP` variable that keeps track of the AP scaling of Serpent sting. Set it to 0 by default but always multiply the AP by it inside of the serpent sting code, then override the value to `0.25` here